modules/templates/templates_api.rb

```diff
@@ -14,8 +14,8 @@ class Proxy::TemplatesApi < Sinatra::Base
   end
 
   get "/:kind" do |kind|
-    log_halt(500, "Failed to retrieve #{kind} template for #{params[:token]}: ") do
-      Proxy::Templates::Handler.get_template(kind, params[:token], params[:static])
+    log_halt(500, "Failed to retrieve #{kind} template for #{params.inspect}: ") do
+      Proxy::Templates::Handler.get_template(kind, params, headers)
```

Comment:
at https://github.com/theforeman/smart-proxy/pull/300#discussion-diff-34666024R18 you are calling `Sinatra::Helpers#headers` to populate headers of the request to Foreman. This method doesn't return incoming request headers, you'll get something like: 
`{"Content-Type"=>nil}`
vs. actual request headers:
`{"HTTP_USER_AGENT"=>"curl/7.32.0", "HTTP_HOST"=>"localhost:8000", "HTTP_ACCEPT"=>"*/*", "HTTP_VERSION"=>"HTTP/1.1"}`